esrally/client/factory.py

```diff
@@ -50,13 +60,15 @@ def __init__(self, hosts, client_options):
             import ssl
 
             self.logger.debug("SSL support: on")
-            self.client_options["scheme"] = "https"
 
             self.ssl_context = ssl.create_default_context(
                 ssl.Purpose.SERVER_AUTH, cafile=self.client_options.pop("ca_certs", certifi.where())
             )
 
-            if not self.client_options.pop("verify_certs", True):
+            # We call get() here instead of pop() in order to pass verify_certs through as a kwarg
+            # to the elasticsearch.Elasticsearch constructor. Setting the ssl_context's verify_mode to
+            # ssl.CERT_NONE is insufficient with version 8.0+ of elasticsearch-py.
```

Comment:
According to https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/config.html#_using_an_sslcontext this is a bug. Can you please report it to the Python client?

Reply:
Will do, for posterity, here's a repro:
```python
from elasticsearch import Elasticsearch
import ssl
import certifi

"""
According to https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/config.html#_using_an_sslcontext
we should be able to omit the kwarg `verify_certs` when creating our client, but in reality this doesn't appear true
"""

ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH, cafile=certifi.where())
ssl_context.check_hostname = False
ssl_context.verify_mode = ssl.CERT_NONE

# works
es = Elasticsearch(
    hosts=["https://localhost:9200"],
    ssl_context=ssl_context,
    verify_certs=False,
    basic_auth=("elastic", "changeme"),
)
print(es.info())

# fails
es = Elasticsearch(
    hosts=["https://localhost:9200"],
    ssl_context=ssl_context,
    basic_auth=("elastic", "changeme"),
)
print(es.info())
```

Reply:
Raised https://github.com/elastic/elasticsearch-py/issues/2167 to track.